decidim-proposals/app/commands/decidim/proposals/admin/answer_proposal.rb

```diff
@@ -41,59 +44,29 @@ def answer_proposal
             form.current_user
           ) do
             attributes = {
-              state: @form.state,
-              answer: @form.answer,
+              state: form.state,
+              answer: form.answer,
               answered_at: Time.current,
-              cost: @form.cost,
-              cost_report: @form.cost_report,
-              execution_period: @form.execution_period
+              cost: form.cost,
+              cost_report: form.cost_report,
+              execution_period: form.execution_period
             }
 
+            attributes[:state_published_at] = Time.current if !initial_has_state_published && form.publish_answer?
+
             proposal.update!(attributes)
           end
         end
 
-        def notify_followers
-          return if (proposal.previous_changes.keys & %w(state)).empty?
-
-          if proposal.accepted?
-            publish_event(
-              "decidim.events.proposals.proposal_accepted",
-              Decidim::Proposals::AcceptedProposalEvent
-            )
-          elsif proposal.rejected?
-            publish_event(
-              "decidim.events.proposals.proposal_rejected",
-              Decidim::Proposals::RejectedProposalEvent
-            )
-          elsif proposal.evaluating?
-            publish_event(
-              "decidim.events.proposals.proposal_evaluating",
-              Decidim::Proposals::EvaluatingProposalEvent
-            )
-          end
-        end
+        def notify_proposal_answer
+          return unless initial_has_state_published || form.publish_answer?
```

Comment:
if the answer should be published do not notify?
```suggestion
          return unless (initial_has_state_published || !form.publish_answer?)
```

Reply:
I think that the logic is right as it is, but it is true that it can be confusing. It should be notified if the answer was already published or if it is going to be published now.

In these cases, I always remember [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws): `!(a || b) == !a && !b`, even when I didn't know its name until now 😄. So, if we want to avoid the `unless` we could write it like this:
```ruby
return if !initial_has_state_published && !form.publish_answer?
```

What do you think about this version?

Reply:
ah! ok, then if the answer was already published it should be notified again.
Yes, I'd rather prefer your suggestion

```
return if !initial_has_state_published && !form.publish_answer?
```